### PR TITLE
Update Validator 0.18

### PR DESCRIPTION
**Pull request by agourlay**
This update our `Validator` infrastructure to the [0.18.1](https://github.com/Keats/validator/blob/master/CHANGELOG.md) version.

The important changes are:
- `#[validate]` becomes `#[validate(nested)]` for fields implementing `Validate`.
- new syntax for validation with custom function
- a validation on an `Option<_>` field will be executed on the contained type if the option is `Some`.

### CODE REVIEW

File: lib/api/src/grpc/validate.rs [hunk 1]
```diff
@@ -40,19 +40,6 @@ where
     }
 }
 
-impl<V> ValidateExt for Vec<V>
```
Comment (agourlay) [inline lib/api/src/grpc/validate.rs:43]: An implement for `Vec<V>` is now in the library creating conflicts.
https://github.com/Keats/validator/blob/0872516333bf365668a22e522c57dac64e60d398/validator/src/traits.rs#L50

File: lib/api/src/grpc/validate.rs [hunk 2]
```diff
@@ -349,19 +336,12 @@ pub fn validate_geo_polygon_line_helper(line: &grpc::GeoLineString) -> Result<()
...
-) -> Result<(), ValidationError> {
-    match line {
-        Some(l) => {
-            if l.points.is_empty() {
-                return Err(ValidationError::new("not_empty"));
-            }
-            validate_geo_polygon_line_helper(l)?;
-            Ok(())
-        }
-        _ => Err(ValidationError::new("not_empty")),
```
Comment (agourlay) [inline lib/api/src/grpc/validate.rs:363]: the None case is checked with an extra validation in https://github.com/qdrant/qdrant/blob/6336395685187dffdcd786e1b3f6f5233541195a/lib/api/src/grpc/conversions.rs#L1475

File: lib/storage/src/rbac/mod.rs [hunk 3]
```diff
@@ -23,13 +23,15 @@ pub enum Access {
...
+
 #[derive(Serialize, Deserialize, Validate, PartialEq, Clone, Debug)]
+#[validate(context = ExistingCollections, mutable)]
 pub struct CollectionAccess {
     /// Collection names that are allowed to be accessed
-    #[validate(custom(
-        function = "validate_unique_collections",
-        arg = "&'v_a mut HashSet<String>"
-    ))]
+    #[validate(custom(function = "validate_unique_collections", use_context))]
```
Comment (agourlay) [inline lib/storage/src/rbac/mod.rs:34]: the new macro refused to work with `HashSet<String>` and forced me to create a new struct.

See context in https://github.com/Keats/validator?tab=readme-ov-file#custom

File: lib/api/src/grpc/validate.rs (hunk 1 above)
Comment (timvisee) [reply lib/api/src/grpc/validate.rs:43]: This is fantastic. This implementation has caused some hidden panics before. :smile:

File: lib/storage/src/rbac/mod.rs (hunk 3 above)
Comment (timvisee) [reply lib/storage/src/rbac/mod.rs:34]: :disappointed: 

Since this struct doesn't leak to other places I'm totally fine with it.

**Pull request merged**